lib/bazel/query.go

```diff
@@ -0,0 +1,113 @@
+package bazel
+
+import (
+	"fmt"
+	"io"
+	"os/exec"
+
+	bpb "github.com/enfabrica/enkit/lib/bazel/proto"
+	"github.com/enfabrica/enkit/lib/proto/delimited"
+
+	"google.golang.org/protobuf/proto"
+)
+
+// streamedBazelCommand exec's out to bazel in the specified workspace with the
+// specified arguments. It is a variable that allows for stubbing during tests.
+//
+// In production, the implementation will return an io.Reader containing stdout,
+// an error channel that will emit any errors (if present) during execution, and
+// an error if any occur while starting the command. errChan is closed after the
+// command completes, but the caller should read all of the returned io.Reader
+// before checking the error channel.
+var streamedBazelCommand = func(cmd *exec.Cmd) (io.Reader, chan error, error) {
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, nil, fmt.Errorf("can't get stdout for bazel query: %v", err)
+	}
+	err = cmd.Start()
+	if err != nil {
+		return nil, nil, fmt.Errorf("can't start bazel command: %v", err)
+	}
+
+	pipeReader, pipeWriter := io.Pipe()
+	errChan := make(chan error)
+	go func() {
```

Comment:
don't fully understand why you use this function. Why not just return stdout, and let the caller consume that pipe directly? stdout itself is io.ReadCloser, why is pipeReader preferred? the error will be delivered on cmd.Wait called by the caller.

Even if we wanted errChan, we could limit go func() to calling cmd.Wait()? and let the reader use stdout directly?